lib/msf/core/modules/loader/base.rb

```diff
@@ -466,9 +466,20 @@ def module_path?(path)
 
   # Tries to determine if a file might be executable,
   def script_path?(path)
-    File.executable?(path) &&
-      !File.directory?(path) &&
-      ['#!', '//'].include?(File.read(path, 2))
+    # warn users if their external modules aren't marked executable
+    # per #14281
+    unless !File.directory?(path) &&
+        ['#!', '//'].include?(File.read(path, 2))
+        return false
+    end
+    if File.executable?(path)
+      return true
+    end
+    unless File.extname(path) == '.rb'
+      # prefer elog since load_error clutters the UI on potential false positives
+      elog("Unable to load module #{path} - LoadError Possible non-executable external module.")
+    end
+    false
```

Comment:
`msftidy.rb` will complain if there's a shebang at the start of a classic module, so I don't think we need to restrict the warning by file extension here (we support external Ruby modules for resource separation like in `auxiliary/dos/smb/smb_slowloris`). We can then combine all the conditions into a single `if` expression to avoid the `unless`-`if`-`unless` whiplash.
```suggestion
    if File.directory?(path) || !['#!', '//'].include?(File.read(path, 2))
      false
    elsif File.executable?(path)
      true
    else
      # prefer elog since load_error clutters the UI on potential false positives
      elog("Unable to load module #{path} - LoadError Possible non-executable external module.")
      false
    end
```

Reply:
Originally I threw in the ext check because I was hitting on all kinds of trash files (`.keep` `.pyc`), but I think that was prior to having the `elog` in the conditional statements.  This makes sense, and is def easier on the eyes!